### PR TITLE
Intrepid2 - cn tet and tri, hcurl in ,hdiv in, hgrad tet comp

### PR DESCRIPTION
**Pull request by kyungjoo-kim**
## Motivation

HGRAD Cn Tet and Tri. I would like to see how PR test reacts for these.

HCURL In all, HDIV In all, HGRAD TET COMP12, HVOL Cn all.

## Testing

On weaver, I see some weird behavior but it could be just my setting.

### CODE REVIEW

**Comment by kyungjoo-kim** [comment]
#8310

File: packages/intrepid2/src/Discretization/Basis/Intrepid2_HGRAD_TET_Cn_FEM.hpp [hunk 1]
```diff
@@ -109,7 +109,7 @@ namespace Intrepid2 {
                          const vinvViewType        vinv );
       };
 
-      template<typename ExecSpaceType, ordinal_type numPtsPerEval,
+      template<typename DevieType, ordinal_type numPtsPerEval,
```
Comment (mperego) [inline packages/intrepid2/src/Discretization/Basis/Intrepid2_HGRAD_TET_Cn_FEM.hpp:112]: Here and below,` DevieType` should be `DeviceType`

File: packages/intrepid2/src/Discretization/Basis/Intrepid2_HGRAD_TET_Cn_FEM_ORTH.hpp [hunk 2]
```diff
@@ -131,7 +131,7 @@ class Basis_HGRAD_TET_Cn_FEM_ORTH {
                const ordinal_type   order);
   };
 
-  template<typename ExecSpaceType, ordinal_type numPtsPerEval,
+  template<typename DevieType, ordinal_type numPtsPerEval,
```
Comment (mperego) [inline packages/intrepid2/src/Discretization/Basis/Intrepid2_HGRAD_TET_Cn_FEM_ORTH.hpp:134]: same here, fix the spelling

**Review by mperego — COMMENTED** [review]
Can you fix the spelling?

**Comment by kyungjoo-kim** [comment]
Ah.  Find replace somehow works fine. I will fix the spell

File: packages/intrepid2/src/Discretization/Basis/Intrepid2_HGRAD_TET_Cn_FEM.hpp [hunk 3]
```diff
@@ -109,7 +109,7 @@ namespace Intrepid2 {
                          const vinvViewType        vinv );
       };
 
-      template<typename DevieType, ordinal_type numPtsPerEval,
+      template<typename Devicetype, ordinal_type numPtsPerEval,
```
Comment (mperego) [inline packages/intrepid2/src/Discretization/Basis/Intrepid2_HGRAD_TET_Cn_FEM.hpp:112]: Now you didn't capitalize the T.  `Devicetype` -> `DeviceType`

File: packages/intrepid2/src/Discretization/Basis/Intrepid2_HGRAD_TET_Cn_FEM_ORTH.hpp [hunk 4]
```diff
@@ -131,7 +131,7 @@ class Basis_HGRAD_TET_Cn_FEM_ORTH {
                const ordinal_type   order);
   };
 
-  template<typename DevieType, ordinal_type numPtsPerEval,
+  template<typename Devicetype, ordinal_type numPtsPerEval,
```
Comment (mperego) [inline packages/intrepid2/src/Discretization/Basis/Intrepid2_HGRAD_TET_Cn_FEM_ORTH.hpp:134]: Need to capitalize the T.  `Devicetype` -> `DeviceType`

**Review by mperego — APPROVED** [review]
Looks good

**Comment by kyungjoo-kim** [comment]
okay. I have a next pr and I including the capital device type in that pr. This just go for merge.

**Pull request merged**